Back-To-School/Assets/BackToSchool/Scripts/GameManagement/GameManager.cs

```diff
@@ -1,86 +1,136 @@
+            _inputManager.Moved    += OnPlayerMove;
+            _inputManager.Rotated  += OnPlayerRotate;
+            _inputManager.Stopped  += OnPlayerStop;
+            _inputManager.Fired    += OnPlayerFire;
+            _inputManager.Reloaded += OnPlayerReloaded;
+            _inputManager.Canceled += OnGameStopped;
+            //InputManager
         }
 
-        private void Update()
+        private void CreateGameInstances()
         {
-            if (Input.GetButtonDown("Cancel"))
-            {
-                if (_isGamePaused)
-                {
-                    ContinueGame();
-                }
-                else
-                {
-                    StopGame();
-                }
-            }
+            _player = Instantiate(_playerPrefab, transform.position, Quaternion.identity);
+            _mainCamera.GetComponent<CameraFollow>().SetTarget(_player.transform);
+            _inputManager = Instantiate(_inputManagerPrefab, transform.position, Quaternion.identity).GetComponent<InputManager>();
+            _inputManager.SetCamera(_mainCamera);
+            _enemySpawner = Instantiate(_enemySpawnerPrefab, transform.position, Quaternion.identity).GetComponent<EnemySpawner>();
+            _enemySpawner.SetTarget(_player);
         }
 
-        private void StopGame()
+        #region PlayerHandlers
+
+        private void OnPlayerReloaded()
         {
-            Time.timeScale = 0f;
-            _PausePanel.SetActive(true);
-            _isGamePaused = true;
+            if (!(_isPlayerDead || _isGamePaused)) _playerShooting.Reload();
         }
 
-        private void ContinueGame()
+        private void OnPlayerFire()
         {
-            Time.timeScale = 1f;
-            _PausePanel.SetActive(false);
-            _isGamePaused = false;
+            if (!(_isPlayerDead || _isGamePaused)) _playerShooting.Fire();
+        }
+
+        private void OnPlayerStop() { _playerMovement.Stop(); }
+
+        private void OnPlayerRotate(RaycastHit rayCastHit)
+        {
+            if (!(_isPlayerDead || _isGamePaused)) _playerMovement.Rotate(rayCastHit);
+        }
+
+        private void OnPlayerMove(Vector3 direction)
+        {
+            if (!(_isPlayerDead || _isGamePaused)) _playerMovement.Move(direction);
         }
 
-        private void GameManager_OnPlayerDeath(object sender, EventArgs args)
+        private void OnPlayerDeath()
         {
+            _isPlayerDead = true;
+            _enemySpawner.SetTarget(null);
             Invoke(nameof(EndGame), _gameOverDelay);
         }
 
-        private void EndGame()
+        #endregion
+
+        #region GameHandlers
+
+        private void StopGame()
+        {
+            Time.timeScale = 0f;
+            _isGamePaused  = true;
+            _presenter.SwitchPausePanel(_isGamePaused);
+        }
+
+        private void OnGameStopped()
         {
-            _gameOverPanel.SetActive(true);
+            if (_isGamePaused) { ContinueGame(); }
```

Comment:
not very good formatting (but sometimes acceptable).
Write better sth like: 
`if (_isGamePaused)
    ContinueGame(); 
else 
    StopGame(); 
`